test/integration/local/test_dynamic_endpoint.py

```diff
@@ -0,0 +1,135 @@
+def volume():
+    try:
+        model_dir = os.path.abspath('test/resources/models')
+        subprocess.check_call(
+            'docker volume create --name dynamic_endpoint_model_volume --opt type=none '
+            '--opt device={} --opt o=bind'.format(model_dir).split())
+        yield model_dir
+    finally:
+        subprocess.check_call('docker volume rm dynamic_endpoint_model_volume'.split())
+
+
+@pytest.fixture(scope='module', autouse=True)
+def container(request, docker_base_name, tag, runtime_config):
+    try:
+        command = (
+            'docker run {}--name sagemaker-tensorflow-serving-test -p 8080:8080'
+            ' --mount type=volume,source=dynamic_endpoint_model_volume,target=/opt/ml/models,readonly'
+            ' -e SAGEMAKER_TFS_DEFAULT_MODEL_NAME=half_plus_three'
+            ' -e SAGEMAKER_TFS_NGINX_LOGLEVEL=info'
+            ' -e SAGEMAKER_BIND_TO_PORT=8080'
+            ' -e SAGEMAKER_SAFE_PORT_RANGE=9000-9999'
+            ' -e SAGEMAKER_TFS_ENABLE_DYNAMIC_ENDPOINT=true'
+            ' {}:{} serve'
+        ).format(runtime_config, docker_base_name, tag)
+
+        proc = subprocess.Popen(command.split(), stdout=sys.stdout, stderr=subprocess.STDOUT)
+
+        attempts = 0
+        while attempts < 5:
+            time.sleep(3)
+            try:
+                requests.get('http://localhost:8080/ping')
+                break
+            except:
+                attempts += 1
+                pass
+
+        yield proc.pid
+    finally:
+        subprocess.check_call('docker rm -f sagemaker-tensorflow-serving-test'.split())
+
+
+def make_invocation_request(data, model_name, content_type='application/json'):
+    headers = {
+        'Content-Type': content_type,
+        'X-Amzn-SageMaker-Custom-Attributes':
+            'tfs-model-name={},tfs-method=predict'.format(model_name)
+    }
+    response = requests.post(INVOCATION_URL, data=data, headers=headers)
+    return json.loads(response.content.decode('utf-8'))
+
+
+def make_load_model_request(data, content_type='application/json'):
+    headers = {
+        'Content-Type': content_type
+    }
+    response = requests.post(MODELS_URL, data=data, headers=headers)
+    return response.content.decode('utf-8')
+
+
+def test_container_start_invocation_fail():
+    x = {
+        'instances': [1.0, 2.0, 5.0]
+    }
+    y = make_invocation_request(json.dumps(x), 'half_plus_three')
+    assert y['error'].startswith('Servable not found for request')
+
+
+def test_load_one_model():
+    model_name = 'half_plus_three'
+    model_data = {
+        'name': model_name,
+        'uri': '/opt/ml/models/half_plus_three'
+    }
+    res = make_load_model_request(json.dumps(model_data))
+    assert res == 'Successfully loaded model {}'.format(model_name)
+
+    x = {
+        'instances': [1.0, 2.0, 5.0]
+    }
+    y = make_invocation_request(json.dumps(x), model_name)
+    assert y == {'predictions': [3.5, 4.0, 5.5]}
+
+
+def test_load_two_models():
```

Comment:
Can we add a test where we predict against a not loaded model? To ensure that is returning the proper response type.

Reply:
Test added.